### PR TITLE
internal/load-version-from-file

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |

### CODE REVIEW

File: dspace/config/clarin-dspace.cfg [hunk 1]
```diff
@@ -246,3 +246,7 @@ file.preview.enabled = false
 ### Storage service ###
 # Synchronization is NOT enabled by default
 sync.storage.service.enabled = true
+
+
+### The build version is stored in the specific file ###
+build.version.file.path =
```
Comment (vidiecan) [inline dspace/config/clarin-dspace.cfg:252]: fill this with a link ponting to a file in dspace/config directory named `VERSION_D`

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/converter/RootConverter.java [hunk 2]
```diff
@@ -29,6 +37,36 @@ public RootRest convert() {
...
+     */
+    private String getBuildVersion() {
+        String buildVersion = "Unknown";
+        String bVersionFilePath = configurationService.getProperty("build.version.file.path");
+
+        if (StringUtils.isBlank(bVersionFilePath)) {
+            return buildVersion;
+        }
+
+        try (FileReader fileReader = new FileReader(bVersionFilePath);
```
Comment (vidiecan) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/converter/RootConverter.java:57]: even if it is filled, check if it exists, if not, do not log anything

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/converter/RootConverter.java [hunk 3]
```diff
@@ -29,6 +37,37 @@ public RootRest convert() {
...
+        }
+
+        try {
+            FileReader fileReader = new FileReader(bVersionFilePath);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+            String line;
+            // Read each line from the file until the end of the file is reached
+            while ((line = bufferedReader.readLine()) != null) {
+                buildVersion = line;
```
Comment (vidiecan) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/converter/RootConverter.java:64]: only one line is read!

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/converter/RootConverter.java (hunk 3 above)
Comment (milanmajchrak) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/converter/RootConverter.java]: Fixed

**Review by vidiecan — APPROVED** [review]

**Review by vidiecan — APPROVED** [review]

**Pull request merged**